### DISCUSSION TITLE
[Question] How to use Spot High-Level API in IsaacSim/Lab?

### DISCUSSION THREAD

**Discussion by Jaraxxus-Me** [isaac-sim/IsaacLab#1267] [Q&A]
### Question

Thanks for the great example about velocity control and low-level joint control for Spot.
Since we don't do low-level control on real Spot, may I ask if it is possible to control the Spot in IsaacSim with High Level Spot API used in BD [examples](https://github.com/boston-dynamics/spot-sdk/tree/master/python/examples)?

Do we have that already somewhere in IsaacLab or we will need to implement custom equivalent low-level controllers?

Thanks!

**Comment by mpgussert** [comment]
Thank you for the suggestion!  We will take this into consideration :) We want to avoid adding too much support for specific robots because we want the platform itself to motivate the creation of that kind of thing. However, It would be super useful if there were some way for users to "pass through" commands from higher APIs automagically. Maybe if we created some kind of base "translator" class that users could then customize to do that conversion? just some thoughts!

**Comment by Jaraxxus-Me** [comment]
Thanks for the quick reply! Yes, a higher-level command interface would be very helpful, especially for Spot, as we usually can't do low-level control. In a real deployment, we currently have to directly use the BD examples instead of verifying that in simulation first.

**Comment by Mayankm96** [comment]
I believe Boston Dynamics provide a research kit with the deployment code: https://bostondynamics.com/reinforcement-learning-researcher-kit/

I advise you to reach out to them to gain access to writing and deploying a learned controller.

**Reply by Jaraxxus-Me** [reply]
Thanks for this information. I guess my problem is that on a real spot, we can (and would like to) only use "Spot Programming", where BD has provided high-level command API. We are wondering if we could do this in the same way in IsaacSim so that there is only a small gap between deployment and simulation.

**Reply by lesurJ** [reply]
Hi @Jaraxxus-Me 
Did you end up with a solution for your problem?
I am facing a similar situation where I would like to simulate Spot in IsaacSim before deploying the code to the real robot, and i am confused on how to achieve this. Thanks!

**Reply by skwirskj** [reply]
I am wondering if anyone had any luck on this issue as well. From what @Mayankm96 linked above, it looks like the researcher kit gives you access to low level controls only, not the higher-level Spot API calls.

**Reply by dipampatel18** [reply]
Hi @Jaraxxus-Me @lesurJ @skwirskj. Did you guys have any luck deploying it on the real spot? I don't think the researcher kit/license API solves the issue since it only gives access to the lower-level leg joints. I have a model trained in sim which doesn't use the higher level functions, so I have no idea how to transfer it to the real robot. Figuring out if my RL agent should learn the initial and final joint positions and time taken (to calculate velocity) and pass that to the SDK to make the sim2real happen. Any thoughts?

**Reply by lesurJ** [reply]
@dipampatel18 for my use case, i can only use high-level API commands and thus did not used Isaac Si/Lab. Sorry :/

**Reply by RandomOakForest** [reply]
Hi Everyone, thank you for following up here. The post is a rather old discussion. Would anybody like to start a new one with updated details and hopefully a description of what you can do in the latest Isaac Lab version?
